state/execution.go

```diff
@@ -278,9 +278,13 @@ func updateValidators(currentSet *types.ValidatorSet, abciUpdates []abci.Validat
 
 	// these are tendermint types now
 	for _, valUpdate := range updates {
+		if valUpdate.VotingPower < 0 {
+			return fmt.Errorf("Voting power can't be negative %v", valUpdate)
+		}
+
 		address := valUpdate.Address
 		_, val := currentSet.GetByAddress(address)
-		if val == nil {
+		if val == nil && valUpdate.VotingPower != 0 {
```

Comment:
If we expect it to be larger than `0` why not check for it instead of unqueality check?

Reply:
👍 will do